src/ActionList/Selection.tsx

```diff
@@ -3,14 +3,18 @@ import {CheckIcon} from '@primer/octicons-react'
 import {ListContext, ActionListProps} from './List'
 import {ActionListItemProps} from './shared'
 import {LeadingVisualContainer} from './Visuals'
+import {ActionListGroupProps, GroupContext} from './Group'
 
 type SelectionProps = Pick<ActionListItemProps, 'selected'>
 export const Selection: React.FC<React.PropsWithChildren<SelectionProps>> = ({selected}) => {
+  const {selectionVariant: groupSelectionVariant} = React.useContext(GroupContext)
   const {selectionVariant: listSelectionVariant} = React.useContext(ListContext)
 
   /** selectionVariant in Group can override the selectionVariant in List root */
   /** fallback to selectionVariant from container menu if any (ActionMenu, SelectPanel ) */
-  const selectionVariant: ActionListProps['selectionVariant'] = listSelectionVariant
+  let selectionVariant: ActionListProps['selectionVariant'] | ActionListGroupProps['selectionVariant']
+  if (typeof groupSelectionVariant !== 'undefined') selectionVariant = groupSelectionVariant
+  else selectionVariant = listSelectionVariant
```

Comment:
```suggestion
  const selectionVariant = groupSelectionVariant ?? listSelectionVariant;
```

Small refactor idea for this line, hope the types line up the same 🤔 When looking at the type here, it seems to match up with the union that exists, e.g. `false | 'single' | 'multiple' | undefined`

Reply:
Fine with me. The above is basically what was in place before, but that's a bit more succinct!